src/main/java/pl/patrycja/ox/ui/InputChecker.java

```diff
@@ -7,13 +7,30 @@
  */
 public class InputChecker {
 
+    private UI ui;
+    private int boardSize;
+
+    public InputChecker(UI ui, int boardSize) {
+        this.ui = ui;
+        this.boardSize = boardSize;
+    }
+
+    public int getNumber() {
+        String input = ui.read();
+        while (!checkIfInteger(input, ui)
+                || !checkIfNumberIsInRange(input, boardSize, ui)) {
+            input = ui.read();
+        }
+        return Integer.parseInt(String.valueOf(input));
+    }
+
     /**
      * This method is called to check if number is Integer. It returns true
      * in case it is an Integer, otherwise returns false.
      *
      * @param number is an argument to check if it is an Integer
      */
-    public static boolean checkNumber(String number, UI ui) {
```

Comment:
Do you still need this documentation if method is private?

Reply:
I will delete it.